spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestPartitionedWrites.java

```diff
@@ -31,11 +34,35 @@
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runners.Parameterized;
 
 public class TestPartitionedWrites extends SparkCatalogTestBase {
+
+  private String targetBranch;
+
   public TestPartitionedWrites(
-      String catalogName, String implementation, Map<String, String> config) {
+      String catalogName, String implementation, Map<String, String> config, String targetBranch) {
     super(catalogName, implementation, config);
+    this.targetBranch = targetBranch;
+  }
+
+  @Parameterized.Parameters(
+      name = "catalogName = {0}, implementation = {1}, config = {2}, branch = {3}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+      {
+        SparkCatalogConfig.SPARK.catalogName(),
+        SparkCatalogConfig.SPARK.implementation(),
+        SparkCatalogConfig.SPARK.properties(),
+        SnapshotRef.MAIN_BRANCH
+      },
+      {
+        SparkCatalogConfig.SPARK.catalogName(),
+        SparkCatalogConfig.SPARK.implementation(),
+        SparkCatalogConfig.SPARK.properties(),
+        "test-branch"
+      }
```

Comment:
I don't think it's necessary to test all the catalogs, especially in these tests which perform writes, and just verify if the data is as expected.